describeconfigs.go

```diff
@@ -0,0 +1,237 @@
+	ErrorCode     int16
+	ErrorMessage  string
+	ResourceType  int8
+	ResourceName  string
+	ConfigEntries []describeConfigsResponseV0ConfigEntry
+}
+
+func (t describeConfigsResponseV0Resource) size() int32 {
+	return sizeofInt16(t.ErrorCode) +
+		sizeofString(t.ErrorMessage) +
+		sizeofInt8(t.ResourceType) +
+		sizeofString(t.ResourceName) +
+		sizeofArray(len(t.ConfigEntries), func(i int) int32 { return t.ConfigEntries[i].size() })
+}
+
+func (t describeConfigsResponseV0Resource) writeTo(wb *writeBuffer) {
+	wb.writeInt16(t.ErrorCode)
+	wb.writeString(t.ErrorMessage)
+	wb.writeInt8(t.ResourceType)
+	wb.writeString(t.ResourceName)
+	wb.writeArray(len(t.ConfigEntries), func(i int) { t.ConfigEntries[i].writeTo(wb) })
+}
+
+func (t *describeConfigsResponseV0Resource) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt16(r, size, &t.ErrorCode); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ErrorMessage); err != nil {
+		return
+	}
+
+	if remain, err = readInt8(r, remain, &t.ResourceType); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ResourceName); err != nil {
+		return
+	}
+	fn := func(r *bufio.Reader, size int) (fnRemain int, fnErr error) {
+		var entry describeConfigsResponseV0ConfigEntry
+		if fnRemain, fnErr = (&entry).readFrom(r, size); err != nil {
+			return
+		}
+		t.ConfigEntries = append(t.ConfigEntries, entry)
+		return
+	}
+	if remain, err = readArrayWith(r, remain, fn); err != nil {
+		return
+	}
+
+	return
+}
+
+//DescribeConfigsResponseV0 descirbe configs response V0 version
+type DescribeConfigsResponseV0 struct {
+	ThrottleTimeMs int32
+	Resources      []describeConfigsResponseV0Resource
+}
+
+func (t DescribeConfigsResponseV0) size() int32 {
+	return sizeofInt32(t.ThrottleTimeMs) +
+		sizeofArray(len(t.Resources), func(i int) int32 { return t.Resources[i].size() })
+}
+
+func (t DescribeConfigsResponseV0) writeTo(wb *writeBuffer) {
+	wb.writeInt32(t.ThrottleTimeMs)
+	wb.writeArray(len(t.Resources), func(i int) { t.Resources[i].writeTo(wb) })
+}
+
+func (t *DescribeConfigsResponseV0) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt32(r, size, &t.ThrottleTimeMs); err != nil {
+		return
+	}
+	fn := func(r *bufio.Reader, size int) (fnRemain int, fnErr error) {
+		var resource describeConfigsResponseV0Resource
+		if fnRemain, fnErr = (&resource).readFrom(r, size); err != nil {
+			return
+		}
+		t.Resources = append(t.Resources, resource)
+		return
+	}
+	if remain, err = readArrayWith(r, remain, fn); err != nil {
+		return
+	}
+
+	return
+}
+
+//DescribeConfig
+type DescribeConfig struct {
+	//ResourceType
+	//0: Unknown
+	//1: Any
+	//2: Topic
+	//3: Group
+	//4: Cluster
+	//5: Broker (new)
+	ResourceType int8
+
+	ResourceName string
+
+	ConfigNames []string
+}
+
+func (c *Conn) describeConfigs(request describeConfigsRequestV0) (*DescribeConfigsResponseV0, error) {
```

Comment:
```suggestion
func (c *Conn) describeConfigs(request describeConfigsRequestV0) (DescribeConfigsResponseV0, error) {
```

As mentioned above, we shouldn't export the struct that corresponds to a particular version of the message.